### PR TITLE
[TOREVIEW] Added --version option

### LINKED ISSUE #29: Display version
User should be able to execute a command to display the version of the program

`statique --version`

pessimiste: 1h30
optimiste: 45min
attendu: 1h

### PR DESCRIPTION
**Pull request by MazieroMarco**
Need to check project structure before merge.
Possibility to add a class for the options ?

Closes #29

### CODE REVIEW

**Comment by LeonardBesseau** [comment]
I saw that coloring was possible with [picocli](https://picocli.info/#_version_info_with_colors) so please add this behaviour before i can accept your PR

**Review by LeonardBesseau — CHANGES_REQUESTED** [review]
See my comment about coloring.

**Comment by LeonardBesseau** [comment]
image
If you want to visualise the colors in intelij

File: src/test/java/ch/heig_vd/app/MainTest.java [hunk 1]
```diff
@@ -10,11 +10,13 @@
...
-     * Rigorous Test :-)
+     * Tests the -v and --version options
      */
     @Test
-    public void shouldAnswerWithTrue()
+    public void shouldDisplayTheCurrentVersion()
     {
-        assertTrue( true );
+        int result = new CommandLine(new Main()).execute("statique", "-v");
+        int result2 = new CommandLine(new Main()).execute("statique", "--version");
```
Comment (LeonardBesseau) [inline src/test/java/ch/heig_vd/app/MainTest.java:19]: It seems it just checks if the program exit normally not that the program displayed anything.
Maybe add a check to verify the program outputted something. (Don't need to check that the version is correct but that is at least outputted version -XXXXXX

File: src/main/java/ch/heig_vd/app/Main.java [hunk 2]
```diff
@@ -4,16 +4,35 @@
...
+        // Parses and executes the options
+        CommandLine commandLine = new CommandLine(new Main());
+        commandLine.parseArgs(args);
+        if (commandLine.isVersionHelpRequested()) { // Show version
+            commandLine.printVersionHelp(System.out);
+            return;
+        }
+
+        // Executes the subcommands
         System.exit(new CommandLine(new Main()).execute(args));
```
Comment (LeonardBesseau) [inline src/main/java/ch/heig_vd/app/Main.java:36]: Do we need to create two command line object ?
Or can we reuse the previous one ?

Comment (MazieroMarco) [reply src/main/java/ch/heig_vd/app/Main.java]: Updated, no additional object is created when executing options and commands.

File: src/test/java/ch/heig_vd/app/MainTest.java (hunk 1 above)
Comment (MazieroMarco) [reply src/test/java/ch/heig_vd/app/MainTest.java:34]: Tests updates, it now checks if the ouput contains the keyword "Statique"

File: src/main/java/ch/heig_vd/app/Main.java (hunk 2 above)
Comment (LeonardBesseau) [reply src/main/java/ch/heig_vd/app/Main.java]: Nice. Good job

File: src/test/java/ch/heig_vd/app/MainTest.java (hunk 1 above)
Comment (LeonardBesseau) [reply src/test/java/ch/heig_vd/app/MainTest.java:34]: Ok. Well done

**Review by LeonardBesseau — APPROVED** [review]
All good. All demands are met

**Pull request merged**